### ISSUE TITLE
Translation not working again

### ISSUE THREAD

**Issue by croco3008** [OpenShot/openshot-qt#1521]
* Operating System: Ubuntu 16.04 x86_64
* OpenShot Version: latest GIT (Openshot daily ppa)

Hi! 

My translation to estonian language has stopped working in latest daily ppa.  The issue appears maybe few month ago. As I remember, there was a similar issue some time ago. I just could not find it from bug list. As I see, Thomas is preparing to release the next version of openshot soon. It's good to fix it before the release will happen.

Here is the relevant log entries:

   language:INFO Attempting to load qt_et_EE.UTF-8 in '/usr/share/qt5/translations'
    language:INFO Attempting to load qt_et in '/usr/share/qt5/translations'
    language:INFO Attempting to load et_EE.UTF-8/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Attempting to load et/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Successfully loaded et/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Qt Detected Languages: ['et', 'en-US', 'en']
    language:INFO LANG Environment Variable: et_EE.UTF-8
    language:INFO LOCALE Environment Variable: et_EE
    language:INFO Skipping English language (no need for translation): et_EE.UTF-8
    language:INFO Skipping English language (no need for translation): et_EE

**Comment by peanutbutterandcrackers** [comment]
@DylanC - I smell something fishy here, Cap'ain!

**Comment by DylanC** [comment]
@croco3008 Can you try v2.4.2 and let us know if you have any success?

**Comment by jpfleury** [comment]
I have a similar problem with OpenShot-v2.4.2-x86_64.AppImage on Xubuntu 17.10 64-bit. In "Edit > Preferences > General", I choose "Français Canadien (fr_CA)" and restart OpenShot, but the interface remains in English, even if "fr_CA" is almost completely translated (94.8%) according to its [translation status](https://translations.launchpad.net/openshot/2.0/+translations).

**Comment by peanutbutterandcrackers** [comment]
@jpfleury - I was unable to reproduce this issue in my Elementary OS 0.4.1 machine with fr_CA. Strange.
![screenshot from 2018-07-04 12-49-02](https://user-images.githubusercontent.com/26764547/42261686-bd0d7e74-7f88-11e8-928b-8973a77e61df.png)

**Comment by peanutbutterandcrackers** [comment]
I just tested it again: changed my default (system/user) locale to different languages and changed OpenShot's language and the language changed. So, it worked.

However, since the language by default is set to 'Default', I would have expected OpenShot's UI language to change to be what the default locale is. Even when it was ne_NP or even Chinese, the default language of OpenShot was still English. I thought a fix had gone in for that...

**Comment by jpfleury** [comment]
@peanutbutterandcrackers See here the interface in English while the selected language is "fr_CA" (OpenShot restarted):

![openshot-fr_ca](https://user-images.githubusercontent.com/687082/42264891-f41303b0-7f3f-11e8-9fc9-6816c8b4e5a8.jpg)

The interface is still in English when selecting "Default" (OpenShot restarted):

![openshot-default](https://user-images.githubusercontent.com/687082/42265046-672a7b3a-7f40-11e8-8e35-0e4f2696f50c.jpg)

even if my locale is not English:

```bash
$ locale
LANG=fr_CA.UTF-8
LANGUAGE=fr_CA:fr_FR:en
LC_CTYPE="fr_CA.UTF-8"
LC_NUMERIC=fr_CA.UTF-8
LC_TIME=fr_CA.UTF-8
LC_COLLATE="fr_CA.UTF-8"
LC_MONETARY=fr_CA.UTF-8
LC_MESSAGES="fr_CA.UTF-8"
LC_PAPER=fr_CA.UTF-8
LC_NAME=fr_CA.UTF-8
LC_ADDRESS=fr_CA.UTF-8
LC_TELEPHONE=fr_CA.UTF-8
LC_MEASUREMENT=fr_CA.UTF-8
LC_IDENTIFICATION=fr_CA.UTF-8
LC_ALL=
```

**Comment by peanutbutterandcrackers** [comment]
I am sorry. But I am still unable to re-create your issue (even with the AppImages - I saw you commented that an AppImage worked for you). I even used the 'Humanity' theme to see if that had any effect. But no. I was able to change the language. Perhaps this issue will need some further testing...
![screenshot from 2018-07-04 14-41-12](https://user-images.githubusercontent.com/26764547/42267631-317f641a-7f99-11e8-8a98-f2a677ae32b8.png)

![screenshot from 2018-07-04 14-46-57](https://user-images.githubusercontent.com/26764547/42267628-306448e8-7f99-11e8-9a86-127254b508de.png)

**Comment by ferdnyc** [comment]
If it's _only_ happening in the PPA, it's a packaging issue. I can't help with that, sorry. I know zip about PPAs.

...Are the language files actually _included_ in the PPA? There was a point where rpmfusion's Fedora RPMs were packaging `openshot-lang` as a separate package from `openshot-qt`, so the translations weren't necessarily installed automatically. Something like that?

**Comment by jpfleury** [comment]
I have this bug with the AppImage *OpenShot-v2.4.2-x86_64.AppImage*.

For the PPA, I don't know, I didn't test it.

**Comment by ferdnyc** [comment]
I just noticed what log message is missing from both the OP's report and @jpfleury 's logs:
```console
    language:INFO Exiting translation system (since we successfully loaded: et)
$ # (Or, in jpfleury's case)
    language:INFO Exiting translation system (since we successfully loaded: fr)
```
The earlier "Successfully loaded _xx_/LC_MESSAGES/OpenShot in '_path_'" message is there, but without that final "Exiting" message it seems possible that OpenShot isn't **applying** the translations.

I'll investigate.

**Comment by ferdnyc** [comment]
I'm worried this was broken by #1332, which intended to fix a problem where _English_ translations weren't being loaded when OpenShot is run in non-English locales, but I fear may have prevented loading of translations for any default locale.

@jpfleury , @croco3008 : Just to check my assumptions, can you tell me if you're able to set OpenShot's interface language to any language that's **not** either English or your system's default? Like, for instance, does OpenShot's language change if you set it to `Italiano (it)`?

**Comment by jpfleury** [comment]
@ferdnyc Yes, I'm able to set OpenShot to any other languages, like Italiano (it).

**Comment by ferdnyc** [comment]
Whoops! Sorry @jpfleury , don't know how I missed the email notification that you'd replied, until just now.

Thanks for testing that. With that confirmation, I'm now convinced that #1332 accidentally broke loading of translation files in the default locale. It should be pretty easy to fix.
